drivers/platform/linux/linux_uart.c

```diff
@@ -0,0 +1,286 @@
+/******************************************************************************/
+
+/**
+ * @struct linux_uart_desc
+ * @brief Linux platform specific UART descriptor
+ */
+struct linux_uart_desc {
+	/** /dev/"device_id" file descriptor */
+	int fd;
+	struct termios *terminal;
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the UART communication peripheral.
+ * @param desc - The UART descriptor.
+ * @param param - The structure that contains the UART parameters.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t uart_init(struct uart_desc **desc, struct uart_init_param *param)
+{
+	struct linux_uart_init_param *linux_init;
+	struct linux_uart_desc *linux_desc;
+	struct uart_desc *descriptor;
+	speed_t speed;
+	char path[64];
+
+	descriptor = malloc(sizeof(*descriptor));
+	if (!descriptor)
+		return FAILURE;
+
+	linux_desc = (struct linux_uart_desc*) malloc(sizeof(struct linux_uart_desc));
+	if (!linux_desc)
+		goto free_desc;
+
+	linux_desc->terminal = (struct termios*) malloc(sizeof(struct termios));
+	if (!linux_desc->terminal)
+		goto free_linux_desc;
+
+	descriptor->extra = linux_desc;
+	linux_init = param->extra;
+
+	snprintf(path, sizeof(path), "/dev/%s", linux_init->device_id);
+
+	linux_desc->fd = open(path, O_RDWR | O_NOCTTY | O_NONBLOCK);
+	if (linux_desc->fd < 0) {
+		printf("%s: Can't open %s\n\r", __func__, path);
+		goto free_terminal;
+	}
+
+	tcgetattr(linux_desc->fd, linux_desc->terminal);
+
+	cfmakeraw(linux_desc->terminal);
+
+	switch(param->baud_rate) {
+		case 50:
+			speed = B50;
+			break;
+		case 75:
+			speed = B75;
+			break;
+		case 110:
+			speed = B110;
+			break;
+		case 134:
+			speed = B134;
+			break;
+		case 150:
+			speed = B150;
+			break;
+		case 200:
+			speed = B200;
+			break;
+		case 300:
+			speed = B300;
+			break;
+		case 600:
+			speed = B600;
+			break;
+		case 1200:
+			speed = B1200;
+			break;
+		case 1800:
+			speed = B1800;
+			break;
+		case 2400:
+			speed = B2400;
+			break;
+		case 4800:
+			speed = B4800;
+			break;
+		case 9600:
+			speed = B9600;
+			break;
+		case 19200:
+			speed = B19200;
+			break;
+		case 38400:
+			speed = B38400;
+			break;
+		default:
+			goto free;
+	}
+	cfsetispeed(linux_desc->terminal, speed);
+	cfsetospeed(linux_desc->terminal, speed);
+
+	linux_desc->terminal->c_cflag &= ~CSIZE;
+	switch(param->size) {
+		case UART_CS_5:
+			linux_desc->terminal->c_cflag |= CS5;
+			break;
+		case UART_CS_6:
+			linux_desc->terminal->c_cflag |= CS6;
+			break;
+		case UART_CS_7:
+			linux_desc->terminal->c_cflag |= CS7;
+			break;
+		case UART_CS_8:
+			linux_desc->terminal->c_cflag |= CS8;
+			break;
+	}
+
+	linux_desc->terminal->c_cflag &= ~PARENB;
+	linux_desc->terminal->c_cflag &= ~PARODD;
+	switch(param->parity) {
```

Comment:
You should add a default, since there are more than two possible parities and we are handling only two for the momment.

Reply:
Done.